test/AutoRest.TestServer.Tests/HeadAsBooleanTrue.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using AutoRest.TestServer.Tests.Infrastructure;
+using HeadAsBooleanTrue;
+using NUnit.Framework;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class HeadAsBooleanTrueTests : TestServerTestBase
+    {
+        public HeadAsBooleanTrueTests(TestServerVersion version) : base(version) { }
+
+        [Test]
+        public Task HttpSuccess200Head() => Test(async (host, pipeline) =>
+        {
+            var response = await new HttpSuccessClient(ClientDiagnostics, pipeline, host).Head200Async();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            Assert.AreEqual(true, response.Value);
+        });
+
+        [Test]
+        public Task HttpSuccess204Head() => Test(async (host, pipeline) =>
+        {
+            var response = await new HttpSuccessClient(ClientDiagnostics, pipeline, host).Head204Async();
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+            Assert.AreEqual(true, response.Value);
+        });
+
+        [Test]
+        public Task HttpSuccess404Head() => Test(async (host, pipeline) =>
```

Comment:
Do we have an endpoint that returns 3xx?

Reply:
We don't have 3xx returns [here](https://github.com/Azure/autorest.testserver/blob/master/swagger/head.json).

Reply:
But we have routes which return 3xx. Do you want me to update the spec to add use case for 3xx?

Reply:
It's ok.